suite/tests/client-interface/features.dll.c

```diff
@@ -71,12 +73,36 @@ read_hw_features(char *feat_str)
     return 1;
 }
 
+/*
+ * FEATURE_PAUTH is identified by six different nibbles. We initially check the API nibble
+ * but on Neoverse V1 hardware this is 0. Running this on a Neoverse V1 machine verifies
+ * that the other nibbles are being checked correctly.
+ */
+void
+check_for_pauth()
+{
+    uint64 id_aa64isar1_el1 = 0;
+
+    asm("mrs %0, ID_AA64ISAR1_EL1" : "=r"(id_aa64isar1_el1));
+
+    ushort gpi = (id_aa64isar1_el1 >> 28) & 0xF;
```

Comment:
Is there a more descriptive name than "gpi" (ditto for gpa, api, apa).  This is hard to understand: where did the 28,24,8,4 come from?  How about an inline comment `28 /* FEAT_FOO*/` or named constant (is DR already exported a named constant?  Or make a new local one.)

Reply:
I have added descriptions to each flag.